src/inspector_agent.cc

```diff
@@ -610,6 +620,14 @@ class NodeInspectorClient : public V8InspectorClient {
     return false;
   }
 
+  bool notifyWaitingForDisconnect() {
+    for (const auto& id_channel : channels_) {
+      if (id_channel.second->notifyWaitingForDisconnect())
```

Comment:
It is not possible to send `executionContextDestroyed` without dirty hacks. We need to intercept every `Runtime.enable` and `Runtime.disable` calls to maintain runtime enabled state on node side. At the same time we can not just calls executionContextDestroyed on inspector since then inspector will mark context as destroyed and any later calls that uses Runtime will return an error.

Based on how much hacks we need to add, how many clients there are in a wild right now, what is probability of using new client and old client at the same time I believe that we can just send nothing here. Protocol definition stated exactly this.

Reply:
I tried to adopt solution that you proposed and it seems like it works!